src/lib.rs

```diff
@@ -80,40 +80,48 @@ pub enum LayoutErr {
     AlignmentNotDivisibleByTwo,
 }
 
+#[derive(Clone, Copy)]
 struct Header {
     len: usize,
     cap: usize,
     alignment: usize,
 }
 
+static DEFAULT_HEADER: Header = Header {
+    len: 0,
+    cap: 0,
+    alignment: 1,
+};
+
 impl<T> MiniVec<T> {
+    #[allow(clippy::clippy::cast_ptr_alignment)]
+    fn is_default(&self) -> bool {
+        self.buf.as_ptr() as *const Header == &DEFAULT_HEADER
```

Comment:
Just in case do not convert pointers directly.
Instead use `as usize`
As far as I know compiler can do tricks with pointer addresses hence it might not be reliable

Reply:
Thanks for the catch! Updated  to use `ptr::eq`.